core/API/Proxy.php

```diff
@@ -374,42 +381,28 @@ public function setHideIgnoredFunctions($hideIgnoredFunctions)
     /**
      * Returns an array containing the values of the parameters to pass to the method to call
      *
+     * @param string $method
      * @param array $requiredParameters array of (parameter name, default value)
      * @param array $parametersRequest
      * @throws Exception
      * @return array values to pass to the function call
      */
-    private function getRequestParametersArray($requiredParameters, $parametersRequest)
+    private function getRequestParametersArray($method, $requiredParameters, $parametersRequest)
     {
         $finalParameters = array();
         foreach ($requiredParameters as $name => $defaultValue) {
-            try {
-                if ($defaultValue instanceof NoDefaultValue) {
-                    $requestValue = Common::getRequestVar($name, null, null, $parametersRequest);
-                } else {
-                    try {
-
-                        if ($name == 'segment' && !empty($parametersRequest['segment'])) {
-                            // segment parameter is an exception: we do not want to sanitize user input or it would break the segment encoding
-                            $requestValue = ($parametersRequest['segment']);
-                        } else {
-                            $requestValue = Common::getRequestVar($name, $defaultValue, null, $parametersRequest);
-                        }
-                    } catch (Exception $e) {
-                        // Special case: empty parameter in the URL, should return the empty string
-                        if (isset($parametersRequest[$name])
-                            && $parametersRequest[$name] === ''
-                        ) {
-                            $requestValue = '';
-                        } else {
-                            $requestValue = $defaultValue;
-                        }
-                    }
-                }
-            } catch (Exception $e) {
+            // copied from Common::getRequestVar. will use default value in case of empty string which appears to be necessary.
+            // (tests fail otherwise)
+            $isValuePresent = isset($parametersRequest[$name]) && (is_array($parametersRequest[$name]) || strlen($parametersRequest[$name]) !== 0);
+
+            $value = $isValuePresent ? $parametersRequest[$name] : $defaultValue;
+            if ($value instanceof NoDefaultValue) {
```

Comment:
maybe this logic + throwing exception could be refactored in a method used here and in `Common::getRequestVar`?